chain/chain/src/chain.rs

```diff
@@ -3462,16 +3465,14 @@ impl<'a> ChainUpdate<'a> {
                 }
             }
         }
-
-        for (shard_id, mut receipt_proofs) in receipt_proofs_by_shard_id {
+        for (_, receipt_proofs) in receipt_proofs_by_shard_id.iter_mut() {
             let mut slice = [0u8; 32];
             slice.copy_from_slice(block.hash().as_ref());
             let mut rng: StdRng = SeedableRng::from_seed(slice);
             receipt_proofs.shuffle(&mut rng);
```

Comment:
can you comment why are we shuffling here?

Reply:
This is the order that the receipts will be processed. We need to sort them so the order is deterministic.